include/vara/Feature/Feature.h

```diff
@@ -21,44 +23,26 @@ namespace vara::feature {
 /// \brief Base class for components of \a FeatureModel.
 class Feature {
 public:
-  using FeatureListType = typename llvm::SmallVector<Feature *, 3>;
-  using RelationshipListTy =
-      typename llvm::SmallVector<std::unique_ptr<Relationship<Feature>>, 3>;
-  using feature_iterator = typename FeatureListType::iterator;
-  using const_feature_iterator = typename FeatureListType::const_iterator;
-  using relationship_iterator = RelationshipListTy ::iterator;
+  using FeatureSetType = typename std::set<Feature *>;
+  using feature_iterator = typename FeatureSetType::iterator;
+  using const_feature_iterator = typename FeatureSetType::const_iterator;
 
-private:
-  string Name;
-  std::optional<Location> Loc;
-  bool Opt;
-  llvm::SmallVector<Feature *, 1> Parents;
-  FeatureListType Children;
-  FeatureListType Excludes;
-  FeatureListType Implications;
-  FeatureListType Alternatives;
-  RelationshipListTy Relationships;
-
-protected:
-  Feature(string Name, bool Opt, std::optional<Location> Loc)
-      : Name(std::move(Name)), Opt(Opt), Loc(std::move(Loc)) {}
+  enum FeatureKind { FK_BINARY, FK_NUMERIC };
```

Comment:
Maybe it would be better to use a enum class here so integer constant can not be confused with the feature kinds.